pkg/ruler/ruler.go

```diff
@@ -444,3 +450,131 @@ func (r *Ruler) newManager(ctx context.Context, userID string) (*promRules.Manag
 	}
 	return promRules.NewManager(opts), nil
 }
+
+// GetRules retrieves the running rules from this ruler and all running rulers in the ring if
+// sharding is enabled
+func (r *Ruler) GetRules(ctx context.Context, userID string) ([]*rules.RuleGroupDesc, error) {
```

Comment:
Can we add a test to at least cover the happy path for this function?

Reply:
👍